Mathlib/RingTheory/FilteredAlgebra/Basic.lean

```diff
@@ -0,0 +1,106 @@
+/-
+Copyright (c) 2024 Nailin Guan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Nailin Guan
+-/
+import Mathlib.Algebra.Module.Defs
+import Mathlib.Algebra.Order.Ring.Int
+import Mathlib.Algebra.GradedMonoid
+import Mathlib.Algebra.Group.Submonoid.Defs
+import Mathlib.Algebra.GradedMulAction
+/-!
+# The filtration on abelian group and ring
+
+In this file, we defined the fitration on abelian group,
+and extend it to get the filtration on ring.
+
+# Main definitions
+
+* `IsFiltration` : For `σ` satisfying `SetLike σ A`, an increasing series of `F` in `σ`
+is filtration if there is another series `F_lt` equal to the supremum of `F` with smaller index
+
+* `IsRingFiltration` : For `σ` satisfying `SetLike σ R` where `R` is a semiring,
+an increasing series `F` in `σ` is ring filtration if `IsFiltration F F_lt` and
+the pointwise multiplication of `F i` and `F j` is in `F (i + j)`
+
+* `IsModuleFiltration` : For `F` satisfying `IsRingFiltration F F_lt` in a semiring `R` and `σM`
+satisfying `SetLike σ M` where `M` is a module over `R`, an increasing series `FM` in `σM` is
+module filtration if `IsFiltration F F_lt` and the pointwise scalar multiplication of
+`F i` and `FM j` is in `F (i +ᵥ j)`
+
+-/
+
+section GeneralFiltration
+
+variable {ι : Type*} [Preorder ι]
+
+variable {A : Type*} {σ : Type*} [SetLike σ A]
```

Comment:
A style nit (here and below):
```suggestion
section GeneralFiltration
variable {ι σ A: Type*}
variable [Preorder ι] [SetLike σ A]
```

Putting all the types first makes it a little easier to use lemmas with `@`. The blank lines are largely a mathport artifact, and are not required by the style guide.